exp/tests/test_study_views.py

```diff
@@ -1143,6 +1144,48 @@ def test_only_one_admin_user_other_admins(self):
         mock_user.groups.all.assert_called_once_with()
 
 
+class StudyDetailViewTestCase(TestCase):
+    def test_model(self) -> None:
+        study_list_view = StudyDetailView()
+        self.assertIs(study_list_view.model, Study)
+
+    def test_permissions(self) -> None:
+        self.assertTrue(
+            issubclass(StudyDetailView, ResearcherLoginRequiredMixin),
+            "StudyDetailView must have ResearcherLoginRequiredMixin",
+        )
+        self.assertTrue(
+            issubclass(StudyDetailView, UserPassesTestMixin),
+            "StudyDetailView must have UserPassesTestMixin",
+        )
+
+    def test_test_func(self):
+        manage_researcher_permissions_view = StudyDetailView()
+        self.assertEqual(
+            manage_researcher_permissions_view.test_func,
+            manage_researcher_permissions_view.user_can_see_or_edit_study_details,
+            "StudyDetailView.test_func must be set to StudyDetailView.user_can_see_or_edit_study_details",
+        )
+
+    @patch.object(StudyDetailView, "request", create=True)
+    @patch.object(SingleObjectMixin, "get_object")
+    def test_user_can_see_or_edit_study_details(
+        self, mock_get_object: Mock, mock_request: Mock,
+    ) -> None:
+        mock_request.user.has_study_perms.return_value = True
+        mock_is_researcher = PropertyMock(return_value=True)
+        type(mock_request.user).is_researcher = mock_is_researcher
+
+        study_detail_view = StudyDetailView()
+
+        self.assertIs(study_detail_view.user_can_see_or_edit_study_details(), True)
+
+        mock_request.user.has_study_perms.assert_called_once_with(
+            StudyPermission.READ_STUDY_DETAILS, mock_get_object()
+        )
+        mock_is_researcher.assert_called_with()
+
+
```

Comment:
Very minor but as we establish patterns for test cases - what we really want to ensure, from a security standpoint, is that we never make changes that let someone who's NOT a researcher, or who IS a researcher but who doesn't have READ_STUDY_DETAILS permissions, can't see this page; and that someone who's a researcher with READ_STUDY_DETAILS permissions on this object can. Can we test that explicitly (along the lines of tests at the top of this file for what happens if you're unauthenticated) or is there a reason (test timing, etc.) we shouldn't be doing that?

Reply:
You nailed it on the head, I'd rather use mocks to quicken up the test suite speed.  What you've request can be done, but I would argue that we can replicate the same behavior from the client by using mocks.  Are you comfortable using mocks to test what would happen if someone attempted to access the view and didn't have the correct permissions?  Or would you rather I use the client?

Reply:
I've added the commit ef7461c to show what I would like to do.

Reply:
Oh, that makes sense, & sounds like a plan - the way the mocks are set up now seems like a good compromise between testing the exact outcome we want and completing the tests in a reasonable amount of time.